src/kouta_indeksoija_service/indexer/kouta/hakukohde.clj

```diff
@@ -30,17 +30,25 @@
   [koulutus]
   (:johtaaTutkintoon koulutus))
 
+(defn- get-alkamiskausi-koodi-uri-from-haku [haku]
+  (get-in haku [:metadata :koulutuksenAlkamiskausi :koulutuksenAlkamiskausiKoodiUri]))
+
 (defn- alkamiskausi-kevat?
   [haku hakukohde]
-  (-> (if (:kaytetaanHaunAlkamiskautta hakukohde) haku hakukohde)
-      :alkamiskausiKoodiUri
-      (clojure.string/starts-with? "kausi_k#")))
+  (let [alkamiskausi-koodi-uri (if (:kaytetaanHaunAlkamiskautta hakukohde)
+                                 (get-alkamiskausi-koodi-uri-from-haku haku)
+                                 (:alkamiskausiKoodiUri hakukohde))]
+    (clojure.string/starts-with? alkamiskausi-koodi-uri "kausi_k#")))
+
+(defn- get-alkamisvuosi-from-haku [haku]
+  (get-in haku [:metadata :koulutuksenAlkamiskausi :koulutuksenAlkamisvuosi]))
 
 (defn- alkamisvuosi
   [haku hakukohde]
-  (-> (if (:kaytetaanHaunAlkamiskautta hakukohde) haku hakukohde)
-      :alkamisvuosi
-      Integer/valueOf))
+  (let [alkamisvuosi (if (:kaytetaanHaunAlkamiskautta hakukohde)
+                       (get-alkamisvuosi-from-haku haku)
```

Comment:
Tässä sama. Tuon metodikutsu on turha ja voisi olla get-in suoraan tässä, niin helpottaa lukemista.